feature/bgp/prefixlimit/otg_tests/bgp_prefix_limit_test/bgp_prefix_limit_test.go

```diff
@@ -483,72 +484,54 @@ func (tc *testCase) verifyPacketLoss(t *testing.T, ate *ondatra.ATEDevice, conf
 		if lossPct >= (100-tolerance) && lossPct <= 100 {
 			t.Logf("Traffic Test Passed! Loss seen as expected: got %v, want 100%% ", lossPct)
 		} else {
-			t.Errorf("Traffic %s is expected to fail: got %v, want 100%% failure", flow.Name(), lossPct)
+			t.Errorf("Traffic %s is expected to fail: got %v, want 100%% failure", flow, lossPct)
 		}
 	}
 }
 
-func captureTrafficStats(t *testing.T, ate *ondatra.ATEDevice, conf gosnappi.Config) {
-	otg := ate.OTG()
-	otgutils.LogPortMetrics(t, otg, conf)
-	ap := ate.Port(t, "port1")
-	aic1 := gnmi.OTG().Port(ap.ID()).Counters()
-	sentPkts := gnmi.Get(t, otg, aic1.OutFrames().State())
-	fptest.LogQuery(t, "ate:port1 counters", aic1.State(), gnmi.Get(t, otg, aic1.State()))
-
-	op := ate.Port(t, "port2")
-	aic2 := gnmi.OTG().Port(op.ID()).Counters()
-	rxPkts := gnmi.Get(t, otg, aic2.InFrames().State())
-	fptest.LogQuery(t, "ate:port2 counters", aic2.State(), gnmi.Get(t, otg, aic2.State()))
-	var lostPkts uint64
-	//account for control plane packets in rxPkts
-	if rxPkts > sentPkts {
-		lostPkts = rxPkts - sentPkts
-	} else {
-		lostPkts = sentPkts - rxPkts
-	}
-	t.Logf("Packets: %d sent, %d received, %d lost", sentPkts, rxPkts, lostPkts)
-
-	if lostPkts > tolerance {
-		t.Logf("Lost Packets: %d", lostPkts)
-	} else {
-		t.Log("Traffic Test Passed!")
-	}
-}
-
 func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, duration time.Duration) {
 	otg := ate.OTG()
 	t.Log("Starting traffic")
 	otg.StartTraffic(t)
 	time.Sleep(duration)
 	otg.StopTraffic(t)
 	t.Log("Traffic stopped")
+	time.Sleep(20 * time.Second)
 }
 
-func configureBGPRoutes(t *testing.T, configElement *config, routeCount uint32) {
+func advertiseBGPRoutes(t *testing.T, conf gosnappi.Config, name string, routes uint32) {
 
 	ate := ondatra.ATE(t, "ate")
 	otg := ate.OTG()
+	if routes == prefixLimit {
+		name = "AtLimit"
+	}
+	cs := gosnappi.NewControlState()
+	cs.Protocol().Route().SetNames([]string{"r4" + name, "r6" + name}).SetState(gosnappi.StateProtocolRouteState.ADVERTISE)
+	otg.SetControlState(t, cs)
 
-	// Modifying the OTG BGP routes configuration
-	configElement.bgpv4RR.Addresses().Clear()
-	configElement.bgpv4RR.Addresses().Add().
-		SetAddress(advertisedRoutesv4Net).
-		SetPrefix(advertisedRoutesv4Prefix).
-		SetCount(uint32(routeCount))
-
-	configElement.bgpv6RR.Addresses().Clear()
-	configElement.bgpv6RR.Addresses().Add().
-		SetAddress(advertisedRoutesv6Net).
-		SetPrefix(advertisedRoutesv6Prefix).
-		SetCount(uint32(routeCount))
+}
 
-	// Modifying the OTG flows
-	configElement.flowV4Incr.SetCount(uint32(routeCount))
-	configElement.flowV6Incr.SetCount(uint32(routeCount))
+func withdrawBGPRoutes(t *testing.T, conf gosnappi.Config, name string, routes uint32) {
 
-	otg.PushConfig(t, configElement.topo)
-	otg.StartProtocols(t)
+	ate := ondatra.ATE(t, "ate")
+	otg := ate.OTG()
+	if routes == prefixLimit {
+		name = "AtLimit"
+	}
+	cs := gosnappi.NewControlState()
+	if name == "all" {
```

Comment:
If we take in a list of names instead we can remove this 'if' statement.

Reply:
updated